apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/ReadRowsAcceptanceTest.cs

```diff
@@ -0,0 +1,137 @@
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Xunit;
+using static Google.Cloud.Bigtable.V2.ReadRowsResponse.Types;
+
+namespace Google.Cloud.Bigtable.V2.Tests
+{
+    public class ReadRowsAcceptanceTest
+    {
+        public static TheoryData<ChunkTestCase> AcceptanceTestCases { get; } = GetAcceptanceTestCases();
+
+        private static TheoryData<ChunkTestCase> GetAcceptanceTestCases()
+        {
+            var typeInfo = typeof(ReadRowsAcceptanceTest).GetTypeInfo();
+            string resourceName = $"{typeInfo.Namespace}.read-rows-acceptance-test.json";
+            using (var reader = new StreamReader(typeInfo.Assembly.GetManifestResourceStream(resourceName)))
+            {
+                var acceptanceTest = JsonConvert.DeserializeObject<AcceptanceTest>(reader.ReadToEnd());
+                var result = new TheoryData<ChunkTestCase>();
+                foreach (var testCase in acceptanceTest.Tests)
+                {
+                    result.Add(testCase);
+                }
+                return result;
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(AcceptanceTestCases))]
+        public async Task Test(ChunkTestCase testCase)
+        {
+            var stream = new MockReadRowsStream(
+                new ReadRowsResponse
+                {
+                    Chunks = { testCase.Chunks.Select(json => CellChunk.Parser.ParseJson(json)) }
+                });
+
+            var responses = new List<Row>();
+            try
+            {
+                // Do not use ToList() here. We want to get all results before the first failure.
+                await stream.AsAsyncEnumerable().ForEachAsync(row => responses.Add(row));
+                Assert.False(testCase.ExpectsError, testCase.Name);
+            }
+            catch (InvalidOperationException)
+            {
+                Assert.True(testCase.ExpectsError, testCase.Name);
+            }
+
+            var results = from row in responses
+                          from family in row.Families
+                          from column in family.Columns
+                          from cell in column.Cells
+                          select new TestResult
+                          {
+                              RowKey = row.Key.ToStringUtf8(),
+                              FamilyName = family.Name,
+                              Qualifier = column.Qualifier.ToStringUtf8(),
+                              Timestamp = cell.TimestampMicros,
+                              Value = cell.Value.ToStringUtf8(),
+                              Label = cell.Labels.FirstOrDefault() ?? "",
+                              Error = false
+                          };
+            var expectedResults = testCase.GetNonExceptionResults();
+            Assert.True(expectedResults.SequenceEqual(results), testCase.Name);
```

Comment:
Assert.Equals(...) works fine on sequences. SequenceEqual is not required.

Reply:
Unfortunately, `Assert.Equals` doesn't allow a failure message to be provided, so if there's a failure on the build/CI machines, there would be no way to know which case failed.